### DISCUSSION TITLE
Can I get the URL as a query by path?

### DISCUSSION THREAD

**Discussion by eai04191** [vercel/vercel#5027] [Help] [answered]
# TL;DR
`https://my-awesome-app.eai.vercel.app/api/https://example.com`
Is it possible to put the URL in the path like this and get it as a query?

---
I have found two ways to get the path string as a query. The path segment and routes in config.
https://vercel.com/docs/serverless-functions/introduction#path-segments
https://vercel.com/docs/configuration#project/routes

Path Segments look nice. I decided to give it a try.
Created the file `/api/[url].ts`.
It's simple.

```ts
import { NowRequest, NowResponse } from "@vercel/node";

export default function (req: NowRequest, res: NowResponse) {
    const { url = null } = req.query;
    res.send(`URL: ${url}`);
}
```

When I access `http://localhost:3001/api/hello`, I got `URL: hello`. As expected. What about the URL?

`http://localhost:3001/api/https://example.com`, I got a 301 redirect to `/api/https:/example.com`.
vercel recognizes the string `//` as a path. I searched for a way to stop this, but I couldn't find one.

Next I tried routes.

```json
{
    "version": 2,
    "routes": [{ "src": "/api/(?<url>.*)", "dest": "/api/index?url=$url" }]
}
```
Similarly, I got a 301 redirect.

So is there any way to solve this problem?

**Comment by styfle** [comment, marked as answer]
The double slash in the path is not a valid URL so it will automatically be redirected to the single slash prior to your function being invoked. There is no way to disable this behavior at this time.

I would suggest changing your API from `http://localhost:3001/api/https://example.com` to `http://localhost:3001/api/https/example.com` or just `http://localhost:3001/api/example.com`
